### ISSUE TITLE
网络中使用fluid.layers.resize_bilinear层之后，在模型预测阶段会将输出都resize到训练阶段的patch大小

### ISSUE THREAD

**Issue by Overtown** [PaddlePaddle/PaddleFormers#109]
如题，我的网络中使用了resize层进行上采样，在模型预测阶段，发现模型会将图像resize到训练时patch大小，模型输出图像不是原始图像分辨率。
但是模型中不使用该层，那么预测阶段模型输出就是原始图像分辨率。
请问fluid.layers.resize_bilinear该层为什么会是这样，是否是resize层的bug。

**Comment by Steffy-zxf** [comment]
感谢反馈！方便贴出相关代码吗？根据上面的描述，我们无法了解到您的问题。

**Comment by Overtown** [comment]
ok,我可以给出我的模型定义，中间使用了resize层上采样。如下：

import paddle.fluid as fluid
import cv2, numpy
import os

class Canny():
    def __init__(self, output_channels, base_channels):
        
        self.out_c = output_channels
        self.base_channels = base_channels


    def ConvLayer(self, input_data, out_c, kernel_size, stride):
        # only conv not include relu
        padding_size = kernel_size // 2 
        x = fluid.layers.pad2d(input=input_data, paddings=[padding_size, padding_size, padding_size, padding_size], mode='reflect')

        x = fluid.layers.conv2d(
                input=x, 
                num_filters=out_c,
                filter_size=kernel_size,
                stride=stride,
                padding=0,
                groups=1,
                act=None,
                bias_attr=False,
                name=None)
        
        return x


    def ResidualBlock(self, input_data, channels, kernel_size, stride):
        norm = 'instance_norm' 
        x_conv1 = self.ConvLayer(input_data, channels, kernel_size, stride)
        x_in1 = norm_layer(x_conv1, norm)
        x_relu = fluid.layers.relu(x_in1)
        x_conv2 = self.ConvLayer(x_relu, channels, kernel_size, stride)
        x_in2 = norm_layer(x_conv2, norm)
        x_out = x_in2 + input_data
        
        return x_out


    def ConvInReluLayer(self, input_data, channels, kernel_size, stride):
        norm = 'instance_norm'
        x_conv = self.ConvLayer(input_data, channels, kernel_size, stride)
        x_in = norm_layer(x_conv, norm)
        x_relu = fluid.layers.relu(x_in)

        return x_relu


    def Net(self, input_data):
        
        # Encoder network
        x = self.ConvInReluLayer(input_data, self.base_channels, kernel_size=9, stride=1)  
        x = self.ConvInReluLayer(x, 2*self.base_channels, kernel_size=3, stride=2)  
        x = self.ConvInReluLayer(x, 2*2*self.base_channels, kernel_size=3, stride=2)  
       
        # Residual Network
        for i in range(5):
            x = self.ResidualBlock(x, 2*2*self.base_channels, kernel_size=3, stride=1)
       
        # Decoder Network
        x = fluid.layers.resize_bilinear(x, scale=2.0)        
        x = self.ConvLayer(x, 2*self.base_channels, kernel_size=3, stride=1)
        x = fluid.layers.relu(x)
        
        x = fluid.layers.resize_bilinear(x, scale=2.0)        
        x = self.ConvLayer(x, self.base_channels, kernel_size=3, stride=1)
        x = fluid.layers.relu(x)
        
        x = self.ConvLayer(x, self.out_c, kernel_size=9, stride=1)
        x = fluid.layers.sigmoid(x)
        
        return x

def norm_layer(input, norm_type='batch_norm', name=None, is_test=False):
    #print 'norm:', norm_type
    if norm_type == 'batch_norm':
        param_attr = fluid.ParamAttr(
            name = None if name is None else name + '_w', initializer=fluid.initializer.Constant(1.0))
        bias_attr = fluid.ParamAttr(
            name = None if name is None else name + '_b', initializer=fluid.initializer.Constant(value=0.0))
        return fluid.layers.batch_norm(
            input,
            param_attr=param_attr,
            bias_attr=bias_attr,
            is_test=is_test,
            moving_mean_name=None if name is None else name + '_mean',
            moving_variance_name=None if name is None else name + '_var')

    elif norm_type == 'instance_norm':
        helper = fluid.layer_helper.LayerHelper("instance_norm", **locals())
        dtype = helper.input_dtype()
        epsilon = 1e-5
        mean = fluid.layers.reduce_mean(input, dim=[2, 3], keep_dim=True)
        var = fluid.layers.reduce_mean(
            fluid.layers.square(input - mean), dim=[2, 3], keep_dim=True)
        if name is not None:
            scale_name = name + "_scale"
            offset_name = name + "_offset"
        else:
            scale_name = None
            offset_name = None
        scale_param = fluid.ParamAttr(
            name=scale_name,
            initializer=fluid.initializer.Constant(1.0),
            trainable=True)
        offset_param = fluid.ParamAttr(
            name=offset_name,
            initializer=fluid.initializer.Constant(0.0),
            trainable=True)
        scale = helper.create_parameter(
            attr=scale_param, shape=input.shape[1:2], dtype=dtype)
        offset = helper.create_parameter(
            attr=offset_param, shape=input.shape[1:2], dtype=dtype)

        tmp = fluid.layers.elementwise_mul(x=(input - mean), y=scale, axis=1)
        tmp = tmp / fluid.layers.sqrt(var + epsilon)
        tmp = fluid.layers.elementwise_add(tmp, offset, axis=1)
        return tmp
    else:
        raise NotImplementedError("norm tyoe: [%s] is not support" % norm_type)

**Comment by Steffy-zxf** [comment]
请问下，您的项目里是用的 哪个module，有没用 PaddleHub的Task类吗？

**Comment by Overtown** [comment]
网络模型是Canny.net()；
没有理解"PaddleHub的Task类"指的是什么。

**Comment by Overtown** [comment]
> 请问下，您的项目里是用的 哪个module，有没用 PaddleHub的Task类吗？
网络模型是Canny.net()；
没有理解"PaddleHub的Task类"指的是什么。

**Comment by Steffy-zxf** [comment]
"PaddleHub的Task类是指 ： https://github.com/PaddlePaddle/PaddleHub/blob/release/v1.1.0/paddlehub/finetune/task.py这里面的Class。
这个Canny.net()是您自己定义的，这里面没有看到用了PaddleHub相关的API。您是怎么用PaddleHub的？

**Comment by Overtown** [comment]
> "PaddleHub的Task类是指 ： [https://github.com/PaddlePaddle/PaddleHub/blob/release/v1.1.0/paddlehub/finetune/task.py这里面的Class。](https://github.com/PaddlePaddle/PaddleHub/blob/release/v1.1.0/paddlehub/finetune/task.py%E8%BF%99%E9%87%8C%E9%9D%A2%E7%9A%84Class%E3%80%82)
> 这个Canny.net()是您自己定义的，这里面没有看到用了PaddleHub相关的API。您是怎么用PaddleHub的？

我还是没有理解你的问题，Canny.net()定义之后，就在session中去feed数据和run了，没有使用什么paddlehub接口

**Comment by Steffy-zxf** [comment]
PaddleHub可以便捷地获取PaddlePaddle生态下的预训练模型，完成模型的管理和一键预测。配合使用Fine-tune API，可以基于大规模预训练模型快速完成迁移学习，让预训练模型能更好地服务于用户特定场景的应用。根据您之前的回复发现，您现在并没有使用PaddleHub，建议以上问题，可以在Paddle（https://github.com/PaddlePaddle/Paddle/issues）
或者models （https://github.com/PaddlePaddle/models/issues）
下提issue。另外，您的项目场景是什么呢？看看是否可以利用PaddleHub解决。

**Comment by Steffy-zxf** [comment]
您好，此issue在近一个星期内暂无更新，我们将于今天内关闭。若在关闭后您仍需跟进提问，可重新开启此问题，
我们将在24小时内回复您。因关闭带来的不便我们深表歉意，请您谅解~感谢您对PaddleHub的支持!
